include/seqan/index/index_fm_rank_dictionary_base.h

```diff
@@ -261,6 +268,91 @@ createRankDictionary(RankDictionary<TValue, TSpec> & dict, TText const & text)
     updateRanks(dict);
 }
 
+template <typename TText, typename TSpec, typename TConfig, typename TOtherText, typename TSA>
+inline void
+createRankDictionary(LF<TText, TSpec, TConfig> & lf, TOtherText const & text, TSA const & sa)
+{
+    typedef typename GetValue<TSA>::Type                    TSAValue;
+    typedef typename Iterator<TSA const, Standard>::Type    TSAIter;
+
+    auto & dict = lf.bwt;
+
+    // Resize the RankDictionary.
+    resize(dict, lengthSum(text) + 1, Exact());
+
+    // Assign the text value by value.
+    TSAIter saItBegin = begin(sa, Standard());
+    TSAIter saItEnd = end(sa, Standard());
+
+    setValue(dict, 0, back(text));
+
+    for (TSAIter saIt = saItBegin; saIt != saItEnd; ++saIt)
+    {
+        TSAValue pos = *saIt;
+
+        if (pos != 0)
+        {
+            setValue(dict, saIt - saItBegin + 1, getValue(text, pos - 1));
+        }
+        else
+        {
+            setValue(dict, saIt - saItBegin + 1, lf.sentinelSubstitute);
+            lf.sentinels = saIt - saItBegin + 1;
+        }
+    }
+
+   // Update all ranks.
+   updateRanks(dict);
+}
+
+template <typename TText, typename TSSetSpec, typename TSpec, typename TConfig, typename TOtherText, typename TSA>
+inline void
+createRankDictionary(LF<StringSet<TText, TSSetSpec>, TSpec, TConfig> & lf, TOtherText const & text, TSA const & sa)
+{
+    typedef typename Value<TSA>::Type                       TSAValue;
+    typedef typename Size<TSA>::Type                        TSize;
+
+    // Resize the RankDictionary.
+    TSize seqNum = countSequences(text);
+    TSize totalLen = lengthSum(text);
+    resize(lf.sentinels, seqNum + totalLen, Exact());
+    resize(lf.bwt, seqNum + totalLen, Exact());
+
+    // Fill the sentinel positions (they are all at the beginning of the bwt).
+    for (TSize i = 0; i < seqNum; ++i)
+    {
+        if (length(text[seqNum - (i + 1)]) > 0)
+        {
+            setValue(lf.bwt, i, back(text[seqNum - (i + 1)]));
+            setValue(lf.sentinels, i, false);
+        }
+    }
+
+    // Compute the rest of the bwt.
+    #pragma omp parallel for
```

Comment:
But this does...

Reply:
I was still cleaning up. 😜 You can now merge that code to your lambda. I just left the pragma there so you know where you have to insert it for lambda. I'll remove it before merging this PR.